chainstate/test-suite/src/tests/nft_issuance.rs

```diff
@@ -0,0 +1,1165 @@
+fn nft_issuance_invalid_name(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut tf = TestFramework::default();
+        let outpoint_source_id = TestBlockInfo::from_genesis(&tf.genesis()).txns[0].0.clone();
+        let mut rng = make_seedable_rng(seed);
+
+        let max_desc_len = tf.chainstate.get_chain_config().token_max_description_len();
+        let max_name_len = tf.chainstate.get_chain_config().token_max_name_len();
+        let max_ticker_len = tf.chainstate.get_chain_config().token_max_ticker_len();
+
+        // try all possible chars for name and ensure everything fails except for alphanumeric chars
+        for c in u8::MIN..u8::MAX {
+            // if c is alphanumeric, then this doesn't produce an error, skip it
+            if c.is_ascii_alphanumeric() {
+                continue;
+            }
+
+            let name = gen_text_with_non_ascii(c, &mut rng, max_name_len);
+
+            let result = tf
+                .make_block_builder()
+                .add_transaction(
+                    TransactionBuilder::new()
+                        .add_input(
+                            TxInput::new(outpoint_source_id.clone(), 0),
+                            InputWitness::NoSignature(None),
+                        )
+                        .add_output(TxOutput::new(
+                            NftIssuanceV1 {
+                                metadata: Metadata {
+                                    creator: random_creator(),
+                                    name,
+                                    description: random_string(&mut rng, 1..max_desc_len)
+                                        .into_bytes(),
+                                    ticker: random_string(&mut rng, 1..max_ticker_len).into_bytes(),
+                                    icon_uri: DataOrNoVec::from(None),
+                                    additional_metadata_uri: DataOrNoVec::from(None),
+                                    media_uri: DataOrNoVec::from(None),
+                                    media_hash: vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 0],
+                                },
+                            }
+                            .into(),
+                            OutputPurpose::Transfer(Destination::AnyoneCanSpend),
+                        ))
+                        .build(),
+                )
+                .build_and_process();
+
+            assert!(matches!(
+                result,
+                Err(ChainstateError::ProcessBlockError(
+                    BlockError::CheckBlockFailed(CheckBlockError::CheckTransactionFailed(
+                        CheckBlockTransactionsError::TokensError(
+                            TokensError::IssueErrorNameHasNoneAlphaNumericChar(_, _)
+                        )
+                    ))
+                ))
+            ));
+        }
+    })
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn issue_test_ticker_too_long(#[case] seed: Seed) {
```

Comment:
Please make all names homogeneous.

No need to state in the function name it's a test. It's already obvious from the macro above.

Reply:
Have a look please, is it ok now or not?